apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-item.js

```diff
@@ -26,6 +26,10 @@ const TemplateSelectorItem = props => {
 		blocks = [],
 	} = props;
 
+	if ( ! blocks.length && 'blank' !== value ) {
```

Comment:
Rather then checking for blocks here, how do you feel about testing `this.state.blocks` in `PageTemplateModal.render()`?

```js
if ( ! this.state.isOpen || isEmpty( this.state.blocks ) ) {
		return null;
}
```

Reply:
I still think the component should function when the props aren't provided as expected. It makes them more resilient when moved to an alternative use case. 

This doesn't preclude your change happening as well.